pkg/cli/admin/catalog/mirror.go

```diff
@@ -285,47 +311,34 @@ func WriteManifests(name, dir string, mapping map[string]string) error {
 	}
 
 	for k, v := range mapping {
-		fromRef, err := imagesource.ParseReference(k)
-		if err != nil {
-			klog.Warningf("error parsing source reference for %s", k)
+		if v.WithDigest == "" {
+			klog.Infof("no digest mapping available for %s, skip writing to ImageContentSourcePolicy", k)
 			continue
 		}
-		toRef, err := imagesource.ParseReference(v)
+		toRef, err := imagesource.ParseReference(v.WithDigest)
 		if err != nil {
-			klog.Warningf("error parsing target reference for %s", v)
+			klog.Warningf("error parsing target reference for %s, skip writing to ImageContentSourcePolicy", v)
 			continue
 		}
 		icsp.Spec.RepositoryDigestMirrors = append(icsp.Spec.RepositoryDigestMirrors, operatorv1alpha1.RepositoryDigestMirrors{
-			Source:  fromRef.Ref.AsRepository().String(),
+			Source:  k,
 			Mirrors: []string{toRef.Ref.AsRepository().String()},
 		})
-
-		// omit digest from target if digest exists
-		to := v
-		if len(toRef.Ref.ID) > 0 {
-			to = toRef.Ref.AsRepository().String()
-		}
-		if _, err := f.WriteString(fmt.Sprintf("%s=%s\n", k, to)); err != nil {
-			return err
-		}
 	}
 
 	// Create an unstructured object for removing creationTimestamp
 	unstructuredObj := unstructured.Unstructured{}
+	var err error
 	unstructuredObj.Object, err = runtime.DefaultUnstructuredConverter.ToUnstructured(&icsp)
 	if err != nil {
-		return fmt.Errorf("ToUnstructured error: %v", err)
+		return nil, fmt.Errorf("ToUnstructured error: %v", err)
```

Comment:
errors always lower case

Reply:
The ICSP generation is verbatim from `oc adm release mirror`, I can update both

Reply:
Yes - please.